dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/metrics/TaskMetrics.java

```diff
@@ -83,11 +85,31 @@ public void incTaskDispatch() {
         taskDispatchCounter.increment();
     }
 
-    public void incTaskInstanceByState(final String state) {
-        if (taskInstanceCounters.get(state) == null) {
+    public void incTaskInstanceByState(final String state, Integer taskInstanceId) {
```

Comment:
```suggestion
    public void incTaskInstanceByState(final String state, String taskInstanceName) {
```

Reply:
Please don't query the database again.

Reply:
Thank you for your suggestion. I will redesign this implementation.